support/ref-docs/src/arrays.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Joins (or concatenates) all arrays passed to this function into one
+ * continuous array.
+ */
+export function joinArrs<T>(...arrs: readonly (readonly T[])[]): T[] {
+  const result: T[] = [];
+
+  for (const arr of arrs) {
+    result.push(...arr);
+  }
+
+  return result;
+}
+
+/**
+ * An implementation of the standardized `flatMap`.
+ *
+ * See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap
+ *
+ * @remarks
+ * Adapted from https://2ality.com/2017/04/flatmap.html
+ */
+export function flatMap<T, U>(
```

Comment:
do you think it makes sense to instead depend on [`array.prototype.flatmap`](https://www.npmjs.com/package/array.prototype.flatmap), which is a polyfill for this API? the advantage is that the dependency can simply be removed the next time we update the minimum node requirement based on the support schedule, rather than having to delete the code itself. also, we'd get bug fixes (if there were any) and other updates for free.